pipeline/DESIGN.md

```diff
@@ -0,0 +1,190 @@
+# Pipeline Architecture
+
+The pipeline architecture allows for defining features over Ceramic data as a series of transformations over that data.
+What follows is a descriptions of each phase of the ceramic pipeline the features it enables and its dependencies.
+
+## Overview
+
+Data enters Ceramic via its API and is stored into the raw_events table.
+From there various transformations are applied producing various intermediate tables.
+Each table schema is considered public API and provides access to kind of query pattern against the data.
+
+```mermaid
+graph LR;
+    raw_events;
+    streams;
+    chain_proofs;
+    conclusion_events;
+    model_schemas;
+    event_states;
+    stream_heads;
+    stream_states;
+
+    raw_events --> conclusion_events;
+    streams --> conclusion_events;
+    chain_proofs --> conclusion_events;
+    conclusion_events --> event_states;
+    model_schemas --> event_states;
+    event_states --> stream_heads;
+    stream_heads --> stream_states;
+```
+
+## raw_events
+
+The raw_events table contains a row for each event in a stream and contains the raw CAR data of the event.
+
+### Features
+
+* Access to the raw event data that can be used to validate signatures
+
+### Schema
+
+| Column     | Type  | Description           |
+| ------     | ----  | -----------           |
+| stream_cid | bytes | Cid of the stream     |
+| event_cid  | bytes | Cid of the event      |
+| car        | bytes | CAR data of the event |
+
+## streams
+
+The streams table contains a row for each stream and contains the dimensions and controller of the stream.
+
+### Features
+
+* Access to the identifying information for streams
+
+### Schema
+
+| Column      | Type              | Description              |
+| ------      | ----              | -----------              |
+| stream_cid  | bytes             | Cid of the stream        |
```

Comment:
Do all the tables in this doc store stream CID or is any stream ID? Or do we just combine stream type + cid on the fly for that?

Reply:
Its always stream cid and you combine with type on the fly to get the stream id. The CID is the unique identifier for the stream. The type is just extra metadata.